backend/core/llm/openai_functions.py

```diff
@@ -50,28 +55,53 @@ def __init__(
         max_tokens: int,
         brain_id: str,
         user_openai_api_key: str,
+        user_openai_api_base: str = None,
+        user_openai_gpt_deployment_id: str = None,
+        user_openai_embedding_deployment_id: str = None,
         # TODO: add streaming
     ) -> "OpenAIFunctionsBrainPicking":  # pyright: ignore reportPrivateUsage=none
         super().__init__(
             model=model,
             chat_id=chat_id,
             max_tokens=max_tokens,
             user_openai_api_key=user_openai_api_key,
+            user_openai_api_base=user_openai_api_base,
+            user_openai_gpt_deployment_id=user_openai_gpt_deployment_id,
+            user_openai_embedding_deployment_id=user_openai_embedding_deployment_id,
             temperature=temperature,
             brain_id=str(brain_id),
             streaming=False,
         )
 
     @property
     def openai_client(self) -> ChatOpenAI:
-        return ChatOpenAI(
-            openai_api_key=self.openai_api_key
-        )  # pyright: ignore reportPrivateUsage=none
+        # Reference: langchain/chat_models/azure_openai.py
+        # We need use AzureChatOpenAI if we are using Azure Functions
+        if self.openai_api_type == "azure":
+            return AzureChatOpenAI(
+                openai_api_key=self.openai_api_key,
+                # to support Azure OpenAI Service custom endpoints
+                openai_api_base=self.openai_api_base,
+                openai_api_type=self.openai_api_type,
+                openai_api_version="2023-07-01-preview",
```

Comment:
Shouldn't this be using the `OPENAI_API_VERSION` env var?

Reply:
Rebase the code and use the OPENAI_API_VERSION env var.